pkg/csi/service/wcp/controller.go

```diff
@@ -1237,149 +1239,228 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 		}
 		volumeType = prometheus.PrometheusBlockVolumeType
 
-		var isVADeleted bool
-		volumeAttachment, err := commonco.ContainerOrchestratorUtility.GetVolumeAttachment(ctx, req.VolumeId, req.NodeId)
-		if err != nil {
-			if apierrors.IsNotFound(err) {
-				log.Infof("VolumeAttachments object not found for volume %q & node %q. "+
-					"Thus, assuming the volume is detached.", req.VolumeId, req.NodeId)
-				isVADeleted = true
-			} else {
-				log.Errorf("failed to retrieve volumeAttachment from API server Err: %v", err)
-				return nil, csifault.CSIInternalFault, err
+		csiDetachEnabled := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx,
+			common.CSIDetachOnSupervisor)
+
+		if csiDetachEnabled {
+			var isVADeleted bool
+			var vmuuid string
+			volumeAttachment, err := commonco.ContainerOrchestratorUtility.GetVolumeAttachment(ctx, req.VolumeId, req.NodeId)
```

Comment:
Will this check to verify that VA has been deleted go away sometime in the future? We don't do this check in other flavors like vanilla & pvcsi where CSI has always had the responsibility  to detach.